### DISCUSSION TITLE
Bandwidth monitoring would be nice

### DISCUSSION THREAD

**Discussion by natrys** [flo-at/minmon#24] [Ideas]
Haven't seen this in roadmap. Currently using monit for egress check up.

**Comment by flo-at** [comment]
The metric to monitor here would be bits/second for each networking device, right?

**Reply by natrys** [reply]
Yeah but useful would be aggregate ingress/egress use during given time interval/cycles. For example in monit I do something like:

```
check network public with interface ens4
  if total uploaded > 100 MB in last day then alert
  if total downloaded > 100 MB in last day then alert
```

I am not sure where this information is best found, but on googling it seems one idea is to parse `/proc/net/dev` (though this is linux specific). Taking snapshot twice and calculating the diff is probably enough.

**Reply by flo-at** [reply]
Okay, so if I understand correctly that's a moving window sum for the last 24 hours that is checked at every check interval. Would be cool to have something like a generic "filter" option to transform the metric into a moving window sum/average, peak, ...

That's three tasks then:
1. Implement network device throughput (input / output)
2. Implement filter infrastructure
3. Implement moving window sum

**Comment by flo-at** [comment]
The filter infrastructure is implemented in #28. There is also a "Sum" filter which I didn't commit yet because none of the existing checks can use it and it would be unused code. The "Sum" filter will be able to sum up the incoming or outgoing data over a configurable number of intervals (i.e. period of time) . There are more filters implemented already like Average and Peak.

What's missing is the actual network data input/output check. It's on my list for the next release.

Thanks a lot for the idea! I didn't really think about filters before but they are very useful for metrics that have some kind of noise (or fluctuation).

**Comment by flo-at** [comment]
A network device rx/tx throughput check is implemented in #53 and will be the next PR to be merged.

I will update the filter implementation after the PR is merged and implement the sum filter at the end.

**Comment by flo-at** [comment]
This has been resolved already and should work as planned.
